pyro/ops/stats.py

```diff
@@ -342,3 +344,63 @@ def waic(input, log_weights=None, pointwise=False, dim=0):
     elpd = lpd - p_waic
     waic = -2 * elpd
     return (waic, p_waic) if pointwise else (waic.sum(), p_waic.sum())
+
+
+def fit_generalized_pareto(X):
+    """
+    Given a dataset X assumed to be drawn from the Generalized Pareto
+    Distribution, estimate the distributional parameters k, sigma using a
+    variant of the technique described in reference [1]. This
+    implementation is modified from the implementation used in reference [2].
+
+    References
+    [1] 'A new and efficient estimation method for the generalized Pareto distribution.'
+    Zhang, J. and Stephens, M.A. (2009).
+    [2] 'Pareto Smoothed Importance Sampling.'
+    Aki Vehtari, Andrew Gelman, Jonah Gabry
+    https://github.com/avehtari/PSIS/blob/904146236767182270c9718a7b2a30831fe701fe/py/psis.py#L211
+
+    :param torch.Tensor or numpy.ndarray X: the input data X
+    :returns tuple: tuple of floats (k, sigma) corresponding to the fit parameters
+    """
+    if isinstance(X, torch.Tensor):
+        X = X.data.cpu().float().numpy()
+
+    if X.ndim != 1 or len(X) <= 1:
+        raise ValueError("Invalid input array.")
+
+    X.sort()
+
+    N = len(X)
+    PRIOR = 3
+    M = 30 + int(math.sqrt(N))
+
+    bs = np.arange(1, M + 1, dtype=float) - 0.5
```

Comment:
i was doing float for simplicity so i don't have variable dtype logic downstream